pkg/prometheus/promcfg.go

```diff
@@ -2078,3 +2082,134 @@ func (cg *ConfigGenerator) GenerateAgentConfiguration(
 
 	return yaml.Marshal(cfg)
 }
+
+func (cg *ConfigGenerator) appendScrapeConfigs(
+	slices []yaml.MapSlice,
+	scrapeConfigs map[string]*v1alpha1.ScrapeConfig,
+	apiserverConfig *v1.APIServerConfig,
+	store *assets.Store,
+	shards int32) []yaml.MapSlice {
+	scrapeConfigIdentifiers := make([]string, len(scrapeConfigs))
+	i := 0
+	for k := range scrapeConfigs {
+		scrapeConfigIdentifiers[i] = k
+		i++
+	}
+
+	// Sorting ensures, that we always generate the config in the same order.
+	sort.Strings(scrapeConfigIdentifiers)
+
+	for _, identifier := range scrapeConfigIdentifiers {
+		slices = append(slices,
+			cg.WithKeyVals("scrapeconfig", identifier).generateScrapeConfig(
+				scrapeConfigs[identifier],
+				apiserverConfig,
+				store,
+				shards,
+			),
+		)
+	}
+
+	return slices
+}
+
+func (cg *ConfigGenerator) generateScrapeConfig(
+	sc *v1alpha1.ScrapeConfig,
+	apiserverConfig *v1.APIServerConfig,
+	store *assets.Store,
+	shards int32,
+) yaml.MapSlice {
+	jobName := fmt.Sprintf("scrapeconfig/%s/%s", sc.Namespace, sc.Name)
+	cfg := yaml.MapSlice{
+		{
+			Key:   "job_name",
+			Value: jobName,
+		},
+	}
+
+	cpf := cg.prom.GetCommonPrometheusFields()
+	labeler := namespacelabeler.New(cpf.EnforcedNamespaceLabel, cpf.ExcludedFromEnforcement, false)
```

Comment:
the namespace labeler uses the Resource` field from `ObjectReference` which is an enum and would need to allow `scrapeconfigs` too. But looking at the code, it would require some refactoring (e.g. moving the resourceToKind outside of the v1 package) which can be done after.

https://github.com/prometheus-operator/prometheus-operator/blob/06b5c4189f3f72737766d86103d049115c3aff48/pkg/apis/monitoring/v1/types.go#L86

Reply:
Sounds good, keeping it as a follow up

Reply:
#5544